README.md

```diff
@@ -195,7 +197,47 @@ new CspHtmlWebpackPlugin({
   processFn: defaultProcessFn  // defined in the plugin itself
 })
 ```
+## Advanced Usage
+### Dumping the CSP directives into a file
+
+Sometimes the `<meta>` tag is not compatible with specific directives, for example `report-uri` or `report-to`.
+For that specific case, it's possible to set your own `processFn` callback to add a conf file that will be used by the reverse-proxy.
+
+Here is an example for nginx in webpack.config.js : 
+```js
+const RawSource = require('webpack-sources').RawSource;
+
+function generateNginxHeaderFile(
+  builtPolicy,
+  _htmlPluginData,
+  _obj,
+  compilation
+) {
+  const header =
+    'add_header Content-Security-Policy "' +
+    builtPolicy +
+    '; report-uri /csp-report/ ";';
+  compilation.emitAsset('nginx-csp-header.conf', new RawSource(header));
+}
 
+module.exports = {
+  {...},
+  plugins: [
+    new CspHtmlWebpackPlugin(
+      {...}, {
+      processFn: generateNginxHeaderFile
+    })
+  ]
+};
+```
```

Comment:
_Rewording slightly to be more consistent with the rest of the readme - do this work for you?_


## Advanced Usage
### Generating a file containing the CSP directives

Some specific directives require the CSP to be sent to the client via a response header (e.g. `report-uri` and `report-to`)

You can set your own `processFn` callback to make this happen.

#### nginx

In your webpack config:
```js
const { RawSource } = require('webpack-sources');

function customProcessFn(builtPolicy, _htmlPluginData, _obj, compilation) {
  const header = `add_header Content-Security-Policy "${builtPolicy}; report-uri /csp-report/ ";`;
  compilation.emitAsset("nginx-csp-header.conf", new RawSource(header));
}

module.exports = {
  // rest of webpack config

  plugins: [
    new CspHtmlWebpackPlugin(
      { ... },
      {
        processFn: customProcessFn,
      }
    ),
  ],
};
```

In your nginx configuration :
```nginx
location / {
  ...
  include /path/to/webpack/output/nginx-csp-header.conf
}
```

Reply:
Updated